tests/e2e/e2e.go

```diff
@@ -65,57 +65,60 @@ func pingSvc(url string) {
 // waitForCmdOut runs a command until it gets
 // the expected output.
 // It accepts 2 arguments, cmd (command to be run)
-// & expOut (expected output).
+// timeout (the time to wait for the output)
+// check (function with output check logic)
 // It times out if the command doesn't fetch the
-// expected output  within the timeout period (1m).
-func waitForCmdOut(cmd string, expOut string) bool {
+// expected output  within the timeout period.
+func waitForCmdOut(cmd string, timeout int, check func(output string) bool) bool {
 
-	pingTimeout := time.After(1 * time.Minute)
+	pingTimeout := time.After(time.Duration(timeout) * time.Minute)
 	tick := time.Tick(time.Second)
 
 	for {
 		select {
 		case <-pingTimeout:
-			Fail("Timeout out after 1 minute")
+			Fail("Timeout out after " + string(timeout) + " minutes")
 
 		case <-tick:
 			out, err := exec.Command("/bin/sh", "-c", cmd).Output()
 			if err != nil {
 				Fail(err.Error())
 			}
 
-			if string(out) == expOut {
+			if check(string(out)) {
```

Comment:
return check(string(out)) ?

Reply:
We don't want to return return false because that would interrupt the test. Either timeout occurs and we exit or the command execution fails or the check condition is met else keep looping.